Assets/_BForBoss/_Core/Scripts/Environment/PlayerTriggerAreaEffect.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using Perigon.Character;
+using Perigon.Entities;
+using Perigon.Utility;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace BForBoss
+{
+    [RequireComponent(typeof(Collider))]
+    public class PlayerTriggerAreaEffect : MonoBehaviour
+    {
+        private enum HealthEffect
+        {
+            Damage,
+            Heal
+        }
+
+        private enum SpeedModifier
+        {
+            SlowDown,
+            SpeedUp
+        }
+        
+        
+        [Serializable]
+        private class PlayerMutation
+        {
+            [Title("Player Effects", "Effect on Player over time", TitleAlignments.Split)]
+            [SerializeField, Min(0.0f), Tooltip("Duration between player getting effected on")] private float _secondsBetweenEffect = 1.0f;
+            [Title("Health Effect", horizontalLine: false)]
+            [SerializeField] private HealthEffect _playerHealthEffect = HealthEffect.Damage;
+            [SerializeField, Min(0), Tooltip("The amount to change the player's health by")] private int _healthChangeAmount = 0;
+
+            [Title("Player Modifications", "Immediate modification on Player's stats", TitleAlignments.Split)]
+            [Title("Speed Modification", horizontalLine: false)]
+            [SerializeField, Tooltip("Effect on Player's max walk speed")] private SpeedModifier _playerSpeedModification = SpeedModifier.SlowDown;
+            [SerializeField, Range(0.0f, 1.0f), Tooltip("modifer to multiply player's speed by")] private float _speedMultiplier = 0f;
+
+            public float secondsBetweenEffect => _secondsBetweenEffect;
+            public HealthEffect playerHealthEffect => _playerHealthEffect;
+            public int healthChangeAmount => _healthChangeAmount;
+
+            public SpeedModifier playerSpeedModification => _playerSpeedModification;
+            public float speedMultiplier => _speedMultiplier;
+        }
+        
+        [SerializeField] private PlayerMutation playerMutation;
+
+        private PlayerLifeCycleBehaviour _playerLifeCycle;
+        private PlayerMovementBehaviour _playerMovementBehaviour;
+        private float _elapsedTime;
+
+        private void Awake()
+        {
+            _elapsedTime = playerMutation.secondsBetweenEffect;
+        }
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (_playerLifeCycle != null && _playerMovementBehaviour != null)
+            {
+                return;
+            }
+
+            _playerLifeCycle = other.GetComponent<PlayerLifeCycleBehaviour>();
+            _playerMovementBehaviour = other.GetComponent<PlayerMovementBehaviour>();
+
+            if (_playerLifeCycle == null || _playerMovementBehaviour == null)
+            {
+                return;
+            }
+            
+            EffectPlayer();
+            ModifyPlayerStats();
```

Comment:
EffectPlayer();
ModifyPlayerStats();

Are essentialy the same thing, so you should only have one method that effects the player. Since it's confusing as to why there's two of the same methods being called.

Reply:
Effect - called repeatedly every `_secondsBetweenEffect` seconds.
Modification - called once when you enter in the trigger zone - and reset when you leave

It is way more intuitive to keep those two different concepts separate!

Reply:
In that case can you update the naming of the method to make it clear. One time effect vs repeated effect. And use the same jargon, either use effect or status change. 

Stick with one. Using Different ones implies different meanings